packages/vite/src/node/server/index.ts

```diff
@@ -567,30 +571,56 @@ export async function createServer(
   return server
 }
 
+type ListenOption =
+  | number
+  | {
+      port: number
+      socket: string
+    }
```

Comment:
I think prettier will also allow this:

```suggestion
type ListenOption = number | { port: number; socket: string }
```